### PR TITLE
[MOS-648] Fix USB connection/disconnection detection

### PR DESCRIPTION
**Pull request by Maciej-Mudita**
If there were errors while charging the battery,
e.g. too high temperature, the detection of USB
connection/disconnection failed.

**Your checklist for this pull request**

Make sure that this PR:
- [x] Complies with our guidelines for contributions
- [x] Has unit tests if possible
- [x] Has documentation updated
- [x] Has changelog entry added

### CODE REVIEW

**Review by SP2FET — APPROVED** [review]

File: module-services/service-evtmgr/battery/BatteryController.cpp [hunk 1]
```diff
@@ -155,16 +155,12 @@ void sevm::battery::BatteryController::updateSoC()
...
-    auto newState            = transformChargingState(charger->getChargingStatus());
-
-    if (chargingState == Store::Battery::State::Discharging && newState != Store::Battery::State::Discharging) {
-        service->bus.sendUnicast(std::make_shared<sevm::USBPlugEvent>(sevm::USBPlugEvent::Event::CablePlugged),
-                                 service::name::service_desktop);
-    }
-
-    if (chargingState != Store::Battery::State::Discharging && newState == Store::Battery::State::Discharging) {
-        service->bus.sendUnicast(std::make_shared<sevm::USBPlugEvent>(sevm::USBPlugEvent::Event::CableUnplugged),
+    auto newPlugState = charger->getPlugState();
```
Comment (dawidwojtas-mudita) [inline module-services/service-evtmgr/battery/BatteryController.cpp:158]: What about `const` here? :)

File: module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp [hunk 2]
```diff
@@ -941,4 +941,14 @@ namespace bsp::battery_charger
         LOG_INFO("\tAvgCurrent: %dmA", getAvgCurrent());
         LOG_INFO("\tRawSoC: %d%%", getBatteryLevel().value());
     }
+
+    bool isUsbPlugged()
+    {
+        auto chargerStatus = chargerRead(Registers::CHG_INT_OK);
```
Comment (dawidwojtas-mudita) [inline module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp:947]: Also here you can add `const` :)

**Review by dawidwojtas-mudita — APPROVED** [review]

File: module-services/service-evtmgr/battery/BatteryController.cpp (hunk 1 above)
Comment (Maciej-Mudita) [reply module-services/service-evtmgr/battery/BatteryController.cpp:158]: done

File: module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp (hunk 2 above)
Comment (Maciej-Mudita) [reply module-bsp/board/rt1051/puretx/bsp/battery_charger/battery_charger.cpp:947]: done

**Review by swichu91 — APPROVED** [review]

**Pull request merged**